charts/dns/templates/pdns-deployment.yaml

```diff
@@ -6,65 +6,125 @@ metadata:
+      {{- with .Values.priorityClassName }}
+      priorityClassName: {{ . | quote }}
+      {{- end }}
+      {{- with .Values.dnsPolicy }}
+      dnsPolicy: {{ . }}
+      {{- end }}
       containers:
         - name: pdns
-          image: "{{ .Values.pdns.repository }}:{{ .Values.pdns.tag }}"
+          image: "{{ .Values.pdns.registry }}/{{ .Values.pdns.repository }}:{{ .Values.pdns.tag }}"
           imagePullPolicy: {{ .Values.pdns.pullPolicy }}
+          {{- with .Values.pdns.securityContext }}
           securityContext:
-            allowPrivilegeEscalation: false
-            capabilities:
-              add:
-              - NET_BIND_SERVICE
-              drop:
-              - all
-          resources:
-            {{- toYaml .Values.pdns.resources | nindent 12 }}
-          {{- if or .Values.pdns.httpService.enabled (or .Values.pdns.udpService.enabled .Values.pdns.tcpService.enabled) }}
-          ports:
-          {{- if .Values.pdns.httpService.enabled }}
-          - containerPort: {{ .Values.pdns.httpService.port }}
-            name: http
-            protocol: TCP
-          {{- end }}
-          {{- if .Values.pdns.udpService.enabled }}
-          - containerPort: {{ .Values.pdns.udpService.port }}
-            name: dns-udp
-            protocol: UDP
-          {{- end }}
-          {{- if .Values.pdns.tcpService.enabled }}
-          - containerPort: {{ .Values.pdns.tcpService.port }}
-            name: dns-tcp
-            protocol: TCP
+            {{- toYaml . | nindent 12 }}
           {{- end }}
+          command:
+            - /bin/bash
+            - /entrypoint/script
+            - /usr/local/sbin/pdns_server
+          args:
+            - --launch=gsqlite3
+            - --gsqlite3-database=$(DATABASE_PATH)
+
+            {{- if .Values.service.http.enabled }}
+            - --webserver=yes
+            - --webserver-address=0.0.0.0
+            - --webserver-port={{ .Values.service.http.port }}
+            - --api=yes
+            - --api-key=$(POWERDNS_API_KEY)
+            {{- end }}
+
+            {{- range $arg := .Values.pdns.extraArgs }}
+            - {{ $arg }}
+            {{- end }}
+          {{- if or .Values.service.http.enabled (or .Values.service.udp.enabled .Values.service.tcp.enabled) }}
+          ports:
+            {{- if .Values.service.tcp.enabled }}
+            - name: dns-tcp
+              protocol: TCP
+              containerPort: {{ .Values.service.tcp.port }}
+            {{- end }}
+            {{- if .Values.service.udp.enabled }}
+            - name: dns-udp
+              protocol: UDP
+              containerPort: {{ .Values.service.udp.port }}
+            {{- end }}
+            {{- if .Values.service.http.enabled }}
+            - name: http
+              protocol: TCP
+              containerPort: {{ .Values.service.http.port }}
+            {{- end }}
           {{- end }}
+          env:
+            - name: "DATABASE_PATH"
+              value: /data/db
```

Comment:
If this is hardcoded, why hardcode it here rather than as an arg?

Reply:
Because it is used by the entrypoint script too.

Reply:
Then I'd add a comment for that.